### DISCUSSION TITLE
HAPI support?

### DISCUSSION THREAD

**Discussion by smithara** [SciQLop/speasy#123] [Ideas]
I feel like we might have discussed this before but I forgot. Are there any plans to support access to HAPI servers?

What barriers are there currently? (e.g. considering non-ISTP-CDF data)

I am asking because it feels like the most reasonable way we can make Swarm products available here and across other platforms (see https://vires.services/hapi/). I think there is a similar story for [INTERMAGNET](https://imag-data.bgs.ac.uk/GIN_V1/hapi), and I think there is some plan for a SuperMAG HAPI server.

**Comment by jeandet** [comment]
@smithara, when I looked at it few years ago, the main blocker was the ability to build a hierarchical inventory of products for any server instead of a flat list of products/datasets. I thought a bit about it recently and something I might implement soon as a first step is an HAPI client module inside Speasy without any specific server set up. This would allow later to both use HAPI to integrate new servers or let users make requests like `spz.get_data("hapi://<some_url>/<some_product>")`. 
IIRC, there was also an issue with local cache invalidation, there was no possibility to know if product changed since last request.

So yes I'm willing to support as much as possible HAPI in Speasy and SciQLop ;).
